src/lib/Simplify.hs

```diff
@@ -103,10 +103,17 @@ simplifyDecls (Nest decl rest) = do
   substEnv' <- extendR substEnv $ simplifyDecls rest
   return (substEnv <> substEnv')
 
+
+
 simplifyDecl :: Decl -> SimplifyM SubstEnv
-simplifyDecl (Let NoInlineLet (Bind (name:>_)) expr) = do
+simplifyDecl (Let NoInlineLet (Bind b@(name:>_)) expr) = do
   x <- simplifyStandalone expr
-  withNameHint name $ emitAnn NoInlineLet (Atom x) $> mempty
+  ~(Var (v':>_)) <- withNameHint name $ emitAnn NoInlineLet (Atom x)
+  return $ b@>Rename v'
+simplifyDecl (Let InstanceLet (Bind b@(name:>_)) (Atom atom)) = do
+  x <- substBuilderR atom
+  ~(Var (v':>_)) <- withNameHint name $ emitAnn InstanceLet (Atom x)
+  return $ b@>Rename v'
```

Comment:
Why so?

Reply:
I was trying to avoid inlining instance dictionaries everywhere, blowing up the size of the IR (e.g the `Floating` dictionary with 18 methods!) I ended up having to revert it in fb1a3d62605a68204326d618a6b1a5e6f16a9456 but I'll try it again when I do the safe-names simplify pass.